analysis/topeft_run2/make_cr_and_sr_plots.py

```diff
@@ -568,29 +611,84 @@ def make_cr_fig(h_mc,h_data,unit_norm_bool,axis='process',var='lj0pt',bins=[],gr
+        rax.fill_between(bin_edges_arr,err_ratio_m_syst,err_ratio_p_syst,step='post', facecolor='none', edgecolor='gray', label='Syst err', hatch='////')
+
+        #Now we calculate total uncertainty band
+        err_m_total = summed_mc_val - np.sqrt(summed_mc_sumw2+np.square(syst_err))
+        err_p_total = summed_mc_val + np.sqrt(summed_mc_sumw2+np.square(syst_err))
+        err_ratio_m_total = np.where(summed_mc_val > 0, err_m_total/summed_mc_val, 1)
+        err_ratio_p_total = np.where(summed_mc_val > 0, err_p_total/summed_mc_val, 1)
+
+        err_m_total = np.append(err_m_total, err_m_total[-1])
+        err_p_total = np.append(err_p_total, err_p_total[-1])
+        err_ratio_m_total = np.append(err_ratio_m_total, err_ratio_m_total[-1])
+        err_ratio_p_total = np.append(err_ratio_p_total, err_ratio_p_total[-1])
+
+        #Now we can plot the total uncertainty band
+        ax.fill_between(bins,err_m_total,err_p_total, step='post', facecolor='none', edgecolor='gray', label='Total err', hatch='\\\\')
+        rax.fill_between(bins,err_ratio_m_total,err_ratio_p_total,step='post', facecolor='none', edgecolor='gray', label='Total err', hatch='\\\\')
+
+    elif unblind and not plot_syst_err:
+        print("\n You are unblinding but not plotting systematic error.")
+        ax.fill_between(bins,err_m_stat,err_p_stat, step='post', facecolor='none', edgecolor='gray', label='Stat err', hatch='\\\\')
+        rax.fill_between(bins,err_ratio_m_stat,err_ratio_p_stat,step='post', facecolor='none', edgecolor='gray', label='Stat err', hatch='\\\\')
+
+    elif not unblind and plot_syst_err: #If Asimov data, no need for total error. If systematic errors available, plot that only
+        print("\nSince you are not unblinding, you only will have systematic uncertainty band in your plots")
+        err_m_syst = np.append(err_m_syst,err_m_syst[-1])
+        err_p_syst = np.append(err_p_syst,err_p_syst[-1])
+        err_ratio_m_syst = np.append(err_ratio_m_syst, err_ratio_m_syst[-1])
+        err_ratio_p_syst = np.append(err_ratio_p_syst, err_ratio_p_syst[-1])
+        bin_edges_arr = np.append(bin_edges_arr, [bin_edges_arr[-1] + (bin_edges_arr[-1] - bin_edges_arr[-2])*0.3])
+
+        #just plot the syst uncertainty band
+        ax.fill_between(bin_edges_arr,err_m_syst,err_p_syst, step='post', facecolor='none', edgecolor='gray', label='Syst err', hatch='////')
+        rax.fill_between(bin_edges_arr,err_ratio_m_syst,err_ratio_p_syst,step='post', facecolor='none', edgecolor='gray', label='Syst err', hatch='////')
 
     # Scale the y axis and labels
     ax.autoscale(axis='y')
     ax.set_xlabel(None)
-    rax.set_ylabel('Ratio', loc='center')
+    rax.set_ylabel('Ratio', loc='center',fontsize=17)
     rax.set_ylim(0.5,1.5)
     labels = [item.get_text() for item in rax.get_xticklabels()]
-    labels[-1] = '>500'
+    #labels[-1] = '>500'
```

Comment:
Remove

Reply:
I mistakenly commented it out. I will reinstate it.